addon/mixins/routes/application.js

```diff
@@ -31,6 +31,13 @@ export default Ember.Mixin.create({
         outlet: outlet,
         parentView: parentView
       });
+
+      // Check if another modal exists then give it focus
+      Ember.run.later(function() {
```

Comment:
No worries! The run loop is how Ember groups together executing its internal workings to improve performance. [This post](http://stackoverflow.com/questions/13597869/what-is-ember-runloop-and-how-does-it-work) explaining it in more detail is pretty legendary. `Ember.run.later` is equivalent in function to setTimeout - you use it to delay the execution of some code. A typical use case might be delaying a closing animation or showing the user something for a certain amount of time. It's a dangerous this to use to 'make sure the correct values are set' in cases like that you're better off using `Ember.run.next` (executes something in the next run loop) or observing a property. `Ember.run.sceheduleOnce` can be useful when you want to tun the function at a certain point in the run loop (e.g. `afterRender`, when DOM is ready).